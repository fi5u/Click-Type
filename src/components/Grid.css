.Grid {
    display: inline-block;
    flex-direction: column;
    font-family: 'Courier New', monospace;
}
