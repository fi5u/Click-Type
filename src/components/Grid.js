import './Grid.css'
import React, { PropTypes } from 'react'
import { Button } from 'react-bootstrap'
import GridRow from './GridRow'

const Grid = ({
    activeElement,
    activeRow,
    characterGrid,
    clickButton,
    suggestedWords,
}) => (
    <div
        className="Grid"
    >
        {characterGrid.map((row, iteration) => {
            return (
                <GridRow
                    clickButton={clickButton}
                    isActive={iteration === activeRow}
                    key={`row-${iteration}`}
                >
                    {row.concat(suggestedWords).map((character, charIteration) => {
                        return(
                            <Button
                                className={`GridItem${iteration === activeRow && charIteration === activeElement ? ' GridItem--is-active' : ''}`}
                                key={character}
                                onClick={() => clickButton(character)}
                            >
                                {character}
                            </Button>
                        )
                    })}
                </GridRow>
            )
        })}
    </div>
)

Grid.propTypes = {
    activeElement: PropTypes.number.isRequired,
    activeRow: PropTypes.number.isRequired,
    characterGrid: PropTypes.array.isRequired,
    clickButton: PropTypes.func.isRequired,
    settings: PropTypes.object.isRequired,
    suggestedWords: PropTypes.array.isRequired,
}

export default Grid
