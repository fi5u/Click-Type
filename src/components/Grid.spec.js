/* global it, jest */
import Grid from './Grid'
import React from 'react'
import { shallow } from 'enzyme'

it('renders Grid without crashing', () => {
    shallow(
        <Grid
            activeElement={0}
            activeRow={0}
            characterGrid={[['a','b','c'],['d','e','f'],['g','h','i']]}
            clickButton={jest.fn()}
        />
    )
})
