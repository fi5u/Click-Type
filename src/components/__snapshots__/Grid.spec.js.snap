exports[`test renders Grid correctly 1`] = `
<div
  className="Grid">
  <div
    className="GridRow GridRow--is-active btn-group"
    role="group">
    <button
      className="GridItem GridItem--is-active btn btn-default"
      disabled={false}
      onClick={[Function]}
      type="button">
      a
    </button>
    <button
      className="GridItem btn btn-default"
      disabled={false}
      onClick={[Function]}
      type="button">
      b
    </button>
    <button
      className="GridItem btn btn-default"
      disabled={false}
      onClick={[Function]}
      type="button">
      c
    </button>
    <button
      className="GridItem btn btn-default"
      disabled={false}
      onClick={[Function]}
      type="button">
      abs
    </button>
    <button
      className="GridItem btn btn-default"
      disabled={false}
      onClick={[Function]}
      type="button">
      ace
    </button>
    <button
      className="GridItem btn btn-default"
      disabled={false}
      onClick={[Function]}
      type="button">
      adder
    </button>
  </div>
  <div
    className="GridRow btn-group"
    role="group">
    <button
      className="GridItem btn btn-default"
      disabled={false}
      onClick={[Function]}
      type="button">
      d
    </button>
    <button
      className="GridItem btn btn-default"
      disabled={false}
      onClick={[Function]}
      type="button">
      e
    </button>
    <button
      className="GridItem btn btn-default"
      disabled={false}
      onClick={[Function]}
      type="button">
      f
    </button>
    <button
      className="GridItem btn btn-default"
      disabled={false}
      onClick={[Function]}
      type="button">
      abs
    </button>
    <button
      className="GridItem btn btn-default"
      disabled={false}
      onClick={[Function]}
      type="button">
      ace
    </button>
    <button
      className="GridItem btn btn-default"
      disabled={false}
      onClick={[Function]}
      type="button">
      adder
    </button>
  </div>
  <div
    className="GridRow btn-group"
    role="group">
    <button
      className="GridItem btn btn-default"
      disabled={false}
      onClick={[Function]}
      type="button">
      g
    </button>
    <button
      className="GridItem btn btn-default"
      disabled={false}
      onClick={[Function]}
      type="button">
      h
    </button>
    <button
      className="GridItem btn btn-default"
      disabled={false}
      onClick={[Function]}
      type="button">
      i
    </button>
    <button
      className="GridItem btn btn-default"
      disabled={false}
      onClick={[Function]}
      type="button">
      abs
    </button>
    <button
      className="GridItem btn btn-default"
      disabled={false}
      onClick={[Function]}
      type="button">
      ace
    </button>
    <button
      className="GridItem btn btn-default"
      disabled={false}
      onClick={[Function]}
      type="button">
      adder
    </button>
  </div>
</div>
`;
